### PR TITLE
Improve search relevance

### PR DESCRIPTION
**Pull request by ivarpruijn**
- Makes certain words like 'React' optional for a search, so users get results without having to typo those.
- Provides more relevance to Tracker > Modeling > Taxonomy.

### CODE REVIEW

File: docs/docusaurus.config.js [hunk 1]
```diff
@@ -165,7 +165,16 @@ const config = {
           // Public API key, safe to commit
         apiKey: '57e02ab0593f5338e36b7bff8235a505',
         indexName: 'objectiv',
-        container: 'main'
+        container: 'main',
+        contextualSearch: true,
```
Comment (ivarpruijn) [inline docs/docusaurus.config.js:169]: Already enabled by default, just making it explicit here

File: docs/docusaurus.config.js [hunk 2]
```diff
@@ -165,7 +165,16 @@ const config = {
...
+        searchParameters: {
+          optionalWords: ['react', 'react native', 'angular', 'browser'],
```
Comment (ivarpruijn) [inline docs/docusaurus.config.js:171]: Makes certain keywords optional in a search, so users don't have to type them. We can add more.

File: docs/docusaurus.config.js [hunk 3]
```diff
@@ -165,7 +165,16 @@ const config = {
...
+          optionalFilters: [
+            'section:tracking<score=3>',
+            'section:modeling<score=2>',
+            'section:taxonomy<score=1>',
```
Comment (ivarpruijn) [inline docs/docusaurus.config.js:175]: Provides order of relevance to the various sections, so results in the Tracking section generally show up higher than Modeling, and then Taxonomy.

File: docs/docusaurus.config.js (hunk 2 above)
Comment (sdirosa) [reply docs/docusaurus.config.js:171]: Does this mean that if I search "Logging" or "Plugins" I would get results for any of the SDKs?

**Review by sdirosa — APPROVED** [review]
Looks good to me, left 1 question. 

Shall we test this on staging?

File: docs/docusaurus.config.js (hunk 2 above)
Comment (ivarpruijn) [reply docs/docusaurus.config.js:171]: Yes it would, but with the latest config changes I made they should now be grouped under their respective SDKs.

**Comment by ivarpruijn** [comment]
> Looks good to me, left 1 question.
> 
> Shall we test this on staging?

Definitely, the Docusaurus beta 17 upgrade branch (which includes these changes) is on staging now. Already tested it yesterday, will do some more today.

**Comment by sdirosa** [comment]
> > Looks good to me, left 1 question.
> > Shall we test this on staging?
> 
> Definitely, the Docusaurus beta 17 upgrade branch (which includes these changes) is on staging now. Already tested it yesterday, will do some more today.

Ah sorry, wasn't clear to me as the base branch of the upgrade branch wasn't this one. All good thanks!

**Comment by ivarpruijn** [comment]
Merging, looks good on staging

File: docs/docusaurus.config.js (hunk 2 above)
Comment (sdirosa) [reply docs/docusaurus.config.js:171]: Very nice, I can see that on staging and I think it's a big improvement!

**Pull request merged**